File: typescript/Fast and Slow Pointers/linked_list_loop.ts
/* Definition of ListNode: 
class ListNode {
    val: number; next: ListNode | null;
    constructor(val: number) {
        this.val = val;
        this.next = null;
    }
}    
*/

function linkedListLoop(head: ListNode): boolean {
    let slow: ListNode | null = head;
    let fast: ListNode | null = head;
    // Check both 'fast' and 'fast.next' to avoid null pointer
    // exceptions when we perform 'fast.next' and 'fast.next.next'.
    while (fast !== null && fast.next !== null) {
        slow = slow!.next;
        fast = fast.next.next!;
        if (fast == slow)
            return true;
    }
    return false;
}

File: csharp/Fast and Slow Pointers/LinkedListLoop.cs
using ds.ListNode;

/*
	Definition of ListNode:
	class ListNode
	{
		public ListNode(int val, ListNode next = null)
		{
			Val = val;
			Next = next;
		}

		public int Val { get; set; }

		public ListNode Next { get; set; }
	}
*/

public class Solution
{
	public bool LinkedListLoop(ListNode head)
	{
		ListNode slow = head;
		ListNode fast = head;

		// Check both 'fast' and 'fast.Next' to avoid null pointer
		// exceptions when we perform 'fast.Next' and 'fast.Next.Next'.
		while (fast != null && fast.Next != null)
		{
			slow = slow.Next;
			fast = fast.Next.Next;

			if (fast == slow)
				return true;
		}

		return false;
	}
}